tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs

```diff
@@ -630,4 +638,20 @@ private void UpdateCustomTitleBarButtonsTextColor()
             RotateLayoutButtonText.Foreground = brush;
         }
     }
+
+    private void WindowEx_WindowStateChanged(object sender, WindowState e)
+    {
+        if (e.Equals(WindowState.Normal))
+        {
+            if (Height == FloatingHorizontalBarHeight)
+            {
+                // If we were in collapsed mode, then maximized, then expanded, then restored, be sure to set the height back to the expanded height
+                // (by default, Windows will set us to the collapsed height)
+                if (_viewModel.ShowingExpandedContent && Height == FloatingHorizontalBarHeight)
+                {
+                    Height = Settings.Default.ExpandedWindowHeight;

```

Comment:
>ExpandedWindowHeight [](http://example.com/codeflow?start=46&length=20)

Do we need to store device independent pixels and then scale them to the monitor that the app is on?

Reply:
Good question. I had originally gone down that path, but it turns out Width/Height are all DPI-aware properties, so applying scaling to them was actually doing double scaling and giving incorrect results.

Reply:
Right, I guess I'm concerned as we could be saving them from one monitor and apply them to a different monitor.

Reply:
I showed this to Jason... the Width/Height properties are DPI aware, so no need to apply the scaling ourselves.